client/src/pages/Profile.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Row, Button, Col, Card } from 'react-bootstrap';
+import SolutionCard from '../components/SolutionCard';
+import { User } from '../../../server/models/User';
+import "./Profile.css";
+
+const Profile = () => {
+  const [user, setUser] = useState(null);
+  const [solutions, setSolutions] = useState([]);
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    fetch('/api/users/')
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Failed to fetch user data');
+        }
+        return response.json();
+      })
+      .then(data => {
+        setUser(data);
+      })
+      .catch(error => {
+        console.error('Error fetching user data: ', error);
```

Comment:
Wouldn't this error just be the error thrown in line 16
so we only get the string 'Failed to fetch user data'? Or am I misunderstanding this

Reply:
I'm not quite sure how the responses are set up, but I think there can also be other error messages thrown based on the response received. Right now, the only error is not getting the data, but I'm not sure if we are going to have more errors in the future that need separate cases.